pkg/oc/cli/cmd/rsync/copytar.go

```diff
@@ -203,13 +209,30 @@ func (r *tarStrategy) String() string {
 	return "tar"
 }
 
-func tarRemote(exec executor, sourceDir string, out, errOut io.Writer) error {
+func tarRemote(exec executor, sourceDir string, includes, excludes []string, out, errOut io.Writer) error {
 	glog.V(4).Infof("Tarring %s remotely", sourceDir)
+
+	exclude := []string{}
+	for _, pattern := range excludes {
+		exclude = append(exclude, fmt.Sprintf("--exclude=%s", pattern))
+	}
+
 	var cmd []string
 	if strings.HasSuffix(sourceDir, "/") {
-		cmd = []string{"tar", "-C", sourceDir, "-c", "."}
+		include := []string{"."}
+		include = append(include, includes...)
```

Comment:
@smarterclayton @soltysh in both of these branches I attempt to mimic the behavior of the `rsync` strategy when `--include` is provided, however this does not seem to work with wildcards / patterns. Any time I provide an `--include` when copying *from* a pod *to* my local env, I receive an error [1] unless I specify an exact filename.

Per the `tar` documentation, there is a `--wildcard` flag I can use, but I seem to get the same error regardless.

1.
```
$ oc rsync --strategy=tar --include="*.yaml" --exclude=LICENSE mypod-1-x29xk:/tmp/myplugin/ ./
tar: *.yaml: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
error: error creating remote tar of source directory: command terminated with exit code 2
```

Running the same command above with `loglevel 4` gives me the command being executed as follows:
```
Remote tar command: tar -C /tmp/myplugin/ -c . *.yaml --exclude=LICENSE
```

Reply:
cc @soltysh ^

Reply:
@smarterclayton ^^

Reply:
@smarterclayton friendly ping